homeassistant/components/tahoma/lock.py

```diff
@@ -0,0 +1,90 @@
+"""Support for Tahoma lock."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import ATTR_BATTERY_LEVEL, STATE_LOCKED, STATE_UNLOCKED
+
+from . import DOMAIN as TAHOMA_DOMAIN, TahomaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Tahoma lock."""
+    controller = hass.data[TAHOMA_DOMAIN]["controller"]
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return if so.